File: org.kbinani/BTimer.cs
#if JAVA
//INCLUDE ..\BuildJavaUI\src\org\kbinani\windows\forms\BTimer.java
#else
using System;

namespace org.kbinani.windows.forms {

    public class BTimer : System.Windows.Forms.Timer {
        // root impl of Click event
        #region event impl Tick
        // root impl of Click event is in BTimer
        public BEvent<BEventHandler> tickEvent = new BEvent<BEventHandler>();
        protected override void OnTick( EventArgs e ) {
            base.OnTick( e );
            tickEvent.raise( this, e );
        }
        #endregion

        public void start() {
            base.Start();
        }

        public BTimer()
            : base() {
        }

        public BTimer( System.ComponentModel.IContainer container )
            : base( container ) {
        }

        public void stop() {
            base.Stop();
        }

        public int getDelay() {
            return base.Interval;
        }

        public void setDelay( int value ) {
            base.Interval = value;
        }

        public bool isRunning() {
            return base.Enabled;
        }
    }

}
#endif

File: BuildJavaUI/src/org/kbinani/windows/forms/BTimer.java
package org.kbinani.windows.forms;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.Timer;
import org.kbinani.BEvent;
import org.kbinani.BEventArgs;
import org.kbinani.BEventHandler;

public class BTimer extends Timer implements ActionListener {
    private static final long serialVersionUID = 9174919033610117641L;
    public BEvent<BEventHandler> tickEvent = new BEvent<BEventHandler>();
    
    public BTimer(){
        super( 100, null );
        addActionListener( this );
    }
    
    public void actionPerformed( ActionEvent e ){
        try{
            tickEvent.raise( this, new BEventArgs() );
        }catch( Exception ex ){
            System.err.println( "BTimer#actionPerformed; ex=" + ex );
        }
    }
}
